boot/freeldr/freeldr/lib/peloader.c

```diff
@@ -624,35 +624,41 @@ PeLdrAllocateDataTableEntry(
         FrLdrHeapFree(DataTableEntry, TAG_WLDR_DTE);
         return FALSE;
     }
-    RtlZeroMemory(Buffer, Length);
 
-    DataTableEntry->BaseDllName.Length = Length;
-    DataTableEntry->BaseDllName.MaximumLength = Length;
-    DataTableEntry->BaseDllName.Buffer = PaToVa(Buffer);
+    // Save Buffer, in case of later failure.
+    BaseDllNameBuffer = Buffer;
+
+    RtlZeroMemory(Buffer, Length);
     while (*BaseDllName != 0)
```

Comment:
Instead of the custom copy routine and saving the buffer, just use RtlCopyMemory.

Reply:
Actually that won't work, because it's a poor-man ANSI to UNICODE conversion that is being done here. BaseDllName is a (pointer to) ANSI string, while the allocated and initialized buffer is UNICODE (UTF-16).

Reply:
Options are: either leave it as is or switch to `UNICODE_STRING` and `RtlAnsiStringToUnicodeString()`, if the latter is wanted.

Reply:
Freeldr doesn't include the whole RTL and the unicode -> ANSI tables for obvious reasons of size limitations. So just keep the code as it currently is.